src/main/java/it/gov/pagopa/pu/debtpositions/service/create/receipt/CreateReceiptServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+                                  DebtPositionMapper debtPositionMapper, PrimaryOrgInstallmentService primaryOrgInstallmentService,
+                                  InstallmentUpdateService installmentUpdateService) {
+    this.receiptNoPIIRepository = receiptNoPIIRepository;
+    this.receiptPIIRepository = receiptPIIRepository;
+    this.receiptMapper = receiptMapper;
+    this.organizationService = organizationService;
+    this.brokerService = brokerService;
+    this.debtPositionSyncService = debtPositionSyncService;
+    this.debtPositionMapper = debtPositionMapper;
+    this.primaryOrgInstallmentService = primaryOrgInstallmentService;
+    this.installmentUpdateService = installmentUpdateService;
+  }
+
+  @Override
+  @Transactional
+  public ReceiptDTO createReceipt(ReceiptWithAdditionalNodeDataDTO receiptDTO, String accessToken) {
+    log.info("createReceipt paymentReceiptId[{}} org/nav[{}/{}]", receiptDTO.getPaymentReceiptId(), receiptDTO.getOrgFiscalCode(), receiptDTO.getNoticeNumber());
+
+    //check if the same receipt is already present on DB.
+    // in this case just ignore it since it simply means that the same receipt has been broadcast
+    // to multiple organizations managed by PU
+    ReceiptNoPII receiptInDb = receiptNoPIIRepository.getByPaymentReceiptId(receiptDTO.getPaymentReceiptId());
+    if (receiptInDb != null) {
+      log.info("Receipt with paymentReceiptId[{}] already present in DB id[{}]", receiptDTO.getPaymentReceiptId(), receiptInDb.getReceiptId());
+      receiptDTO.setReceiptId(receiptInDb.getReceiptId());
+      return receiptDTO;
+    }
+
+    //persist receipt
+    Receipt receipt = receiptMapper.mapToModel(receiptDTO);
+    long newId = receiptPIIRepository.save(receipt);
+    receiptDTO.setReceiptId(newId);
+    log.debug("Receipt paymentReceiptId[{}} persisted with id[{}]", receiptDTO.getPaymentReceiptId(), newId);
+
+    //check if organization who handle the notice is managed by PU
+    boolean[] primaryOrgFound = new boolean[]{false};
+    organizationService.getOrganizationByFiscalCode(receiptDTO.getOrgFiscalCode(), accessToken)
+      .ifPresent(primaryOrg -> primaryOrgInstallmentService.findPrimaryOrgInstallment(primaryOrg, receiptDTO.getNoticeNumber())
+        .ifPresent(installment -> {
+          log.debug("primaryOrg installment found id[{}]", installment.getInstallmentId());
+          Broker primaryBroker = brokerService.findById(primaryOrg.getBrokerId(), accessToken);
+          primaryOrgFound[0] = true;
+          //update installment status
+          DebtPosition debtPosition = installmentUpdateService.updateInstallmentStatusOfDebtPosition(installment, primaryBroker, receiptDTO);
+          //start debt position workflow
+          DebtPositionDTO debtPositionDTO = debtPositionMapper.mapToDto(debtPosition);
+          invokeWorkflow(debtPositionDTO, accessToken);
+        }));
```

Comment:
Move these line on a separate class UpdatePaidDebtPositionService
```suggestion
    DebtPositionDTO debtPosition =  organizationService.getOrganizationByFiscalCode(receiptDTO.getOrgFiscalCode(), accessToken)
      .flatMap(primaryOrg -> primaryOrgInstallmentService.findPrimaryOrgInstallment(primaryOrg, receiptDTO.getNoticeNumber()))
      .map(installment -> {
        log.debug("primaryOrg installment found id[{}]", installment.getInstallmentId());
        Broker primaryBroker = brokerService.findById(primaryOrg.getBrokerId(), accessToken);
        primaryOrgFound[0] = true;
        //update installment status
        DebtPosition debtPosition = installmentUpdateService.updateInstallmentStatusOfDebtPosition(installment, primaryBroker, receiptDTO);
        //start debt position workflow
        DebtPositionDTO debtPositionDTO = debtPositionMapper.mapToDto(debtPosition);
        invokeWorkflow(debtPositionDTO, accessToken);
        return debtPositionDTO;
      });
```

Reply:
This has not been resolved

Reply:
done